README.md

```diff
@@ -13,70 +13,258 @@ npm install @shopify/shopify-function-test-helpers
+
+// Load a test fixture
+const fixture = await loadFixture("/path/to/fixtures/my-test.json");
+
+// Build the function
+await buildFunction("/path/to/function");
+
+// Validate the test assets
+const validationResult = await validateTestAssets({
+  schema,
+  fixture,
+  inputQueryAST,
+});
+
+console.log(`  Input Query: ${validationResult.inputQuery.valid ? '✅' : '❌'}`);
+console.log(`  Input Fixture: ${validationResult.inputFixture.valid ? '✅' : '❌'}`);
+console.log(`  Input Query-Fixture Match: ${validationResult.inputQueryFixtureMatch.valid ? '✅' : '❌'}`);
+console.log(`  Output Fixture: ${validationResult.outputFixture.valid ? '✅' : '❌'}`);
+
+// Run the function
+const runResult = await runFunction(
+  fixture.export,
+  fixture.input,
+  "/path/to/function"
+);
+
+console.log("Output:", runResult.result.output);
+console.log("Expected:", fixture.expectedOutput);
+```
+
+### Complete Test Example
+
+For a full test suite that runs multiple fixtures:
+
+```javascript
+import path from "path";
+import fs from "fs";
+import {
   buildFunction,
-  runFunction
-} = require('@shopify/shopify-function-test-helpers');
+  loadFixture,
+  runFunction,
+  validateTestAssets,
+  loadSchema,
+  loadInputQuery,
+} from "@shopify/shopify-function-test-helpers";
+
+describe("Function Tests", () => {
+  let schema;
+  let inputQueryAST;
+  let functionDir;
+
+  beforeAll(async () => {
+    functionDir = path.dirname(__dirname);
+    await buildFunction(functionDir);
+
+    const schemaPath = path.join(functionDir, "schema.graphql");
+    const inputQueryPath = path.join(functionDir, "src/run.graphql");
+
+    schema = await loadSchema(schemaPath);
+    inputQueryAST = await loadInputQuery(inputQueryPath);
+  }, 10000);
+
+  const fixturesDir = path.join(__dirname, "fixtures");
+  const fixtureFiles = fs
+    .readdirSync(fixturesDir)
+    .filter((file) => file.endsWith(".json"))
+    .map((file) => path.join(fixturesDir, file));
+
+  fixtureFiles.forEach((fixtureFile) => {
+    test(`runs ${path.basename(fixtureFile)}`, async () => {
+      const fixture = await loadFixture(fixtureFile);
+
+      const validationResult = await validateTestAssets({
+        schema,
+        fixture,
+        inputQueryAST,
+      });
+
+      expect(validationResult.inputQuery.valid).toBe(true);
+      expect(validationResult.inputFixture.valid).toBe(true);
+      expect(validationResult.inputQueryFixtureMatch.valid).toBe(true);
+      expect(validationResult.outputFixture.valid).toBe(true);
+
+      const runResult = await runFunction(
+        fixture.export,
+        fixture.input,
+        functionDir
+      );
+
+      expect(runResult.error).toBeNull();
+      expect(runResult.result.output).toEqual(fixture.expectedOutput);
+    }, 10000);
+  });
+});
+```
 
-// Load a fixture from test data
-const fixture = loadFixture('20250915_184036_156Z_extensions_cart-checkout-validation_ba711d.json');
+## API Reference
+
+### Core Functions
```

Comment:
WDYT about just linking to the code/doc blocks here? That way it doesn't need to be duplicated and it's always up-to-date.

Reply:
Yes, exactly! But pointing to the code in `main`.